src/effects/VST3/VST3Utils.cpp

```diff
@@ -73,3 +76,70 @@ bool VST3Utils::ParseAutomationParameterKey(const wxString& key, Steinberg::Vst:
    return false;
 
 }
+
+
+
+Steinberg::IPtr<PresetsBufferStream> PresetsBufferStream::fromString(const wxString& str)
+{
+   Steinberg::Buffer buffer(str.size() / 4 * 3);
+   auto len = Base64::Decode(str, buffer);
+   assert(len <= buffer.getSize());
+   buffer.setSize(len);
+
+   auto result = owned(safenew PresetsBufferStream);
+   result->mBuffer.take(buffer);
+   return result;
+}
+
+
+wxString PresetsBufferStream::toString() const
+{
+   return Base64::Encode(mBuffer, mBuffer.getSize());
+}
+
+
+
+bool VST3Wrapper::FetchSettings(VST3EffectSettings& vst3Settings) const
+{
+   using namespace Steinberg;
+
+   auto processorState = owned(safenew PresetsBufferStream);
+   if (mEffectComponent->getState(processorState) == kResultOk)
+   {
+      vst3Settings.mProcessorStateStr = processorState->toString();
+   }     
+
+   auto controllerState = owned(safenew PresetsBufferStream);
+   if (mEditController->getState(controllerState) == kResultOk)
+   {
+      vst3Settings.mControllerStateStr = controllerState->toString();
+   }
+
+   return true;   
+}
+
+
+
+bool VST3Wrapper::StoreSettings(const VST3EffectSettings& vst3settings) const
+{
+   using namespace Steinberg;
+
+   // we need at least the processor state string, otherwise we can not set the EditController
+   if (!vst3settings.mProcessorStateStr)
+      return false;
+
+   auto processorState = PresetsBufferStream::fromString(vst3settings.mProcessorStateStr.value());
+   if (mEffectComponent->setState(processorState) != kResultOk)
+      return false;
+
+   if (vst3settings.mControllerStateStr)
+   {
+      auto controllerState = PresetsBufferStream::fromString(vst3settings.mControllerStateStr.value());
```

Comment:
Remove the two uses of `std::optional::value` above and use the `*` operator instead.

It breaks the Mac build, as do certain other C++17 library functions that can throw new kinds of exceptions.  But on Mac they can't build and link unless a certain minimum version of the operating system is targeted, above our present minimum.

The difference with `*` is that is just gives undefined behavior if there is no value, not an exception.

But you have the proper null (or should I say nullopt) checks before use so it is safe.

Reply:
ok, done

Reply:
The second commit in the sequence is still not buildable.

Rebase and squash the fix onto the second commit.